rs/replicated_state/src/canister_state/system_state.rs

```diff
@@ -1987,7 +1987,10 @@ impl SystemState {
                     }
 
                     // This should not happen. Bail out and let Message Routing deal with it.
-                    Err(_) => return,
+                    Err(e) => {
+                        debug_assert!(false, "Failed to induct message to self: {:?}", e);
+                        return;
+                    }
```

Comment:
Do I understand correctly that on mainnet we can (and will) regularly end up in this path. It is just that in tests our expectation is that we never want a situation where we don't induct and hence we debug assert?

Reply:
> Do I understand correctly that on mainnet we can (and will) regularly end up in this path.

Umm, no. This is an actual critical error case. It can only happen if we get a guaranteed response for a non-existent callback; or if we get any response to an existing callback, but with mismatching respondent / deadline. I.e. if there's a bug.

And we will end up producing a critical error in this case.

`should_enqueue_input()` is technically intended to answer the question "should we enqueue this message or drop it silently?". And the answer should be either `Ok(true)` (enqueue) or `Ok(false)`. `Err(_)` means something went horribly wrong and we should (ideally) panic; or, more practically, raise hell.